src/app/config/or_state_st.h

```diff
@@ -87,6 +87,18 @@ struct or_state_t {
 
   /** When did we last rotate our onion key?  "0" for 'no idea'. */
   time_t LastRotatedOnionKey;
+
+  /** Do we automatically fail IPv4 or IPv6? If so, set one of these to 1. */
+  int IPv4AutoFail;
+  int IPv6AutoFail;
+
+  /** When did we fail IPv4 or IPv6? */
+  time_t IPv4AutoFailTime;
```

Comment:
This name is confusing:
* the variable actually tracks the earliest time in the last week that we failed an IPv4 connection because there was no route to a host.
* but maybe we should track this information over smaller time periods, so we can forget the earliest failures more regularly.